rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java

```diff
@@ -445,6 +445,18 @@ public TransactionInfo getTransactionInfo(byte[] txHash) {
         return txInfo;
     }
 
+    @Override
+    public TransactionInfo getTransactionInfoForMainChainBlock(Transaction tx, Block blockMainChain) {
```

Comment:
```suggestion
    public TransactionInfo getTransactionInfo(Transaction tx, byte[] blockHash) {
```
in this method we don't need entire block object. in some scenarios we might not have a block object to provide but only its hash

Reply:
👍 addressed